Tensile/Source/lib/include/Tensile/ContractionProblem.hpp

```diff
@@ -618,10 +637,35 @@ namespace Tensile
 
         Alpha alpha = static_cast<Alpha>(0);
         Beta  beta  = static_cast<Beta>(0);
+
+        constexpr static uint32_t typeId()
+        {
+            return ContractionInputs::typeId(TypeInfo<A>::Enum,
```

Comment:
Not sure if which is the better way but we should consider an enum class for the type IDs

```cpp
enum class ContractionTypeID: uint32_t
{
    Float = FloatContractionInputs::TypeID(),
    Double = DoubleContractionInputs::TypeID(),
    // ...
}
```
The advantage would be that if there's a type missing in any of the switch statements, there would be a compiler warning (which could be configured as an error).

The disadvantage would be that the switch statements would be a bit less hard to follow:

```cpp
switch(contractionInputsTypeId)
{
    case ContractionTypeID::Float:
    {
        auto const& typedInputs = dynamic_cast<FloatContractionInputs const&>(inputs);
        return solveTyped(problem, typedInputs, hardware);
    }
    // ...
}
```
we've lost the consistency between the case line and the body of the case.

Reply:
When I read it, I do tend to prefer the explicitness of checking the TypedInputs::TypeId() as a reference to the following cast within the case. I think the extra layer of ID's would obfuscate it a bit.